HW_Object_types_with_index_members_and_discriminative_union.ts

```diff
@@ -0,0 +1,47 @@
+// 1. Interface with index signature and union types
+
+interface StringNumberIndex {
+    [key: string]: number | string;
+}
+
+// 2. Interface where values are functions
+
+interface FunctionIndex {
+    [key: string]: (...args: any[]) => any;
+}
+
+// 3. Interface with numeric keys and a specific value type
+
+interface ArrayLike<T> {
+    [index: number]: T;
+}
+
+// 4. Interface with specific properties and an index signature
+
+interface DynamicProperties {
+    name: string;
+    [key: string]: string | number;
+}
+
+// 5. Interfaces with an index signature and extension
+
+interface BasicIndex {
+    [key: string]: number;
+}
+
+interface ExtendedIndex {
```

Comment:
Extended index signature has been used in the requirements that you should use the extend feature between two interfaces.